central/telemetry/phonehome/interceptor.go

```diff
@@ -60,6 +61,16 @@ func getRequestDetails(ctx context.Context, err error, info *grpc.UnaryServerInf
 		method = "unknown"
 		code = -1
 	}
+
+	// By default, all permanent HTTP headers in grpc-gateway are added grpcgateway- prefix:
+	// https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/mux.go#L106-L114
+	// User-Agent header is occupied with internal grpc-go value:
+	// https://github.com/grpc/grpc-go/blob/0238b6e1cec37b55820b461d3d30652c54efe2c4/clientconn.go#L211-L215
```

Comment:
It's actually not clear to me what this code reference proves. `cc.dopts.copts.UserAgent += " " + grpcUA` hints that `User-Agent` value is not overwritten but appended to, so it will be something like `ACS roxctl user agent blah grpc-go/1.50.1`.  Which sort of contradicts what's claimed in ROX-13596.

Could you please split the "Testing Performed" section into before / after?

Reply:
Regarding `cc.dopts.copts.UserAgent`: this shows where the  user agent's value `grpc-go/1.51.0` is initially taken from. 

Answering the question of why there's no other `user-agent` value: 
1. To transform from HTTP to GRPC request, we connect to our gRPC server as a client from within the central
https://github.com/stackrox/stackrox/blob/0529062eb7396c82871e5f59305108638c7b27c0/pkg/grpc/server.go#L352-L353
https://github.com/stackrox/stackrox/blob/0529062eb7396c82871e5f59305108638c7b27c0/pkg/grpc/server.go#L311-L317
2. Connection for this client is created once during central initialization, thus we cannot override it on every request.
3. Value for `user-agent` dial option is not specified in our code, thus we use the default
https://github.com/stackrox/stackrox/blob/0529062eb7396c82871e5f59305108638c7b27c0/pkg/grpc/server.go#L226-L233

Reply:
Ah, I see, didn't know that we persist the connection — thanks. This makes sense.

It does *not* make sense to capture your reply as a code comment because it builds on non-local knowledge which might change any time and render the comment misleading, but it *does* make sense to explain that in the PR description.

One last thing: do you want to follow up and augment grpc server init code with User-Agent for HTTP muxer?

Reply:
> One last thing: do you want to follow up and augment grpc server init code with User-Agent for HTTP muxer?

If I understand correctly what you suggest, that would add some additional `user-agent` value to all requests going through the muxer. Essentially, what @0x656b694d did here:
https://github.com/stackrox/stackrox/pull/3672/files#diff-af62cb3a3a7c777bd1b8feed18657c3ecae951d2dbdea55527f08c8c67967c71R233

I think it works as a fallback option. It's better than displaying `grpc-go/1.51.0`, so why not?

I had a small concern regarding introducing more variety to potential `user-agent` values in Segment, but Segment supports `contains` filter, so that shouldn't be a problem.

Reply:
Yeah, essentially what's done in https://github.com/stackrox/stackrox/pull/3672, so my comment should be irrelevant. Why do we still see `"User-Agent": "grpc-go/1.51.0"` in the Before section then?